pangea-sdk/v3/service/store/integration_test.go

```diff
@@ -0,0 +1,413 @@
+	url := resp.AcceptedResult.PutURL
+
+	fd := pangea.FileData{
+		File: file,
+		Name: "someName",
+	}
+
+	uploader := store.NewFileUploader()
+	err = uploader.UploadFile(ctx, url, pangea.TMputURL, fd)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	var pr *pangea.PangeaResponse[any]
+	i := 0
+
+	for i < 24 {
+		// Wait until result should be ready
+		time.Sleep(time.Duration(10 * time.Second))
+
+		pr, err = client.PollResultByID(ctx, *resp.RequestID, &store.PutResult{})
+		if err == nil {
+			break
+		}
+		i++
+	}
+	assert.NoError(t, err)
+	assert.NotNil(t, pr)
+	assert.NotNil(t, pr.Result)
+
+	_, ok := (*pr.Result).(*store.PutResult)
+	assert.True(t, ok)
+
+}
+
+func Test_Integration_LifeCycle(t *testing.T) {
+	ctx, cancelFn := context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancelFn()
+
+	cfg := pangeatesting.IntegrationConfig(t, testingEnvironment)
+	client := store.New(cfg)
+
+	// Create a folder
+	respCreate, err := client.FolderCreate(ctx, &store.FolderCreateRequest{
+		Path: FOLDER_FILES,
+	})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	assert.NotNil(t, respCreate)
+	assert.NotNil(t, respCreate.Result)
+	assert.NotEmpty(t, respCreate.Result.Object.ID)
+	folderID := respCreate.Result.Object.ID
+
+	// Upload a file with path as unique param
+	file, err := os.Open(PDF_FILEPATH)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	respPut, err := client.Put(ctx,
+		&store.PutRequest{
+			Path:           FOLDER_FILES + "/" + TIME + "_file_multipart_1",
+			TransferMethod: pangea.TMmultipart,
+		},
+		file)
+
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	assert.NotNil(t, respPut)
+	assert.NotNil(t, respPut.Result)
+	assert.Equal(t, folderID, respPut.Result.Object.ParentID)
+	assert.Empty(t, respPut.Result.Object.Metadata)
+	assert.Empty(t, respPut.Result.Object.Tags)
+	assert.Empty(t, respPut.Result.Object.MD5)
+	assert.Empty(t, respPut.Result.Object.SHA512)
+	assert.NotEmpty(t, respPut.Result.Object.SHA256)
+
+	// Upload a file with parent id and name
+	// Upload a file with path as unique param
+	file, err = os.Open(PDF_FILEPATH)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	respPut2, err := client.Put(ctx,
+		&store.PutRequest{
+			Name:           TIME + "_file_multipart_2",
+			ParentID:       folderID,
+			TransferMethod: pangea.TMmultipart,
+			Metadata:       METADATA,
+			Tags:           TAGS,
+		},
+		file)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	assert.NotNil(t, respPut2)
+	assert.NotNil(t, respPut2.Result)
+	assert.Equal(t, folderID, respPut2.Result.Object.ParentID)
+	assert.Equal(t, store.Metadata(METADATA), respPut2.Result.Object.Metadata)
+	assert.Equal(t, store.Tags(TAGS), respPut2.Result.Object.Tags)
+	assert.Empty(t, respPut2.Result.Object.MD5)
+	assert.Empty(t, respPut2.Result.Object.SHA512)
+	assert.NotEmpty(t, respPut2.Result.Object.SHA256)
+
+	// Update file. full metadata and tags
```

Comment:
(nitpick) maybe?:
```suggestion
	// Update file with full metadata and tags.
```
edit: repeats elsewhere below as well

Reply:
fixed